lib/publiccloud/instance.pm

```diff
@@ -350,55 +350,50 @@ sub wait_for_ssh {
     $args{username} //= $self->username();
     $args{public_ip} //= $self->public_ip();
     my $start_time = time();
-    my $check_port = 1;
     my $sleep_period = $args{ignore_wrong_pubkey} ? 20 : 1;
-
+    my $retry = ($args{timeout} > $sleep_period) ? int($args{timeout} / $sleep_period) : 1;
+    my $xaccess;
+    my $out;
+    my $status = '';
     # Looping until reaching timeout or passing two conditions :
     # - SSH port 22 is reachable
-    # - journalctl got message about reaching one of certain targets
-    while ((my $duration = time() - $start_time) < $args{timeout}) {
-        if ($check_port) {
-            $check_port = 0 if (script_run('nc -vz -w 1 ' . $self->{public_ip} . ' 22', quiet => 1) == 0);
+    # - is-system-running ok
+    $xaccess = ssh_script_retry('nc -vz -w 1 ' . $self->{public_ip} . ' 22', timeout => $args{timeout}, delay => $sleep_period, retry => $retry);
+    if ($xaccess == 0) {
+        # helps in boottime tests,when hard reboot could change the instance address:
+        ssh_script_run("ssh-keyscan $args{public_ip} | tee -a ~/.ssh/known_hosts") if (get_var('PUBLIC_CLOUD_PERF_COLLECT') or get_var('PUBLIC_CLOUD_CHECK_BOOT_TIME'));
```

Comment:
wait but if we got here there is no problem with ssh keys , isn't this call needs to be done in `else` branch ?

Reply:
This command is in the ok branch of [master version](https://github.com/m-dati/os-autoinst-distri-opensuse/blob/8dcf744faa8c124b55b217378a3bf505bb09f126/lib/publiccloud/instance.pm#L365), probably due to some old test IP-changed issue

Reply:
ok than